java/fury-core/src/main/java/io/fury/codegen/Expression.java

```diff
@@ -182,12 +182,16 @@ public ExprCode doGenCode(CodegenContext ctx) {
           hasCode = true;
         }
       }
-      ExprCode lastExprCode = last.genCode(ctx);
       String code = codeBuilder.toString();
       if (!hasCode) {
         code = null;
       }
-      return new ExprCode(code, lastExprCode.isNull(), lastExprCode.value());
+      if (code != null) {
+        ExprCode lastExprCode = last.genCode(ctx);
+        return new ExprCode(code, lastExprCode.isNull(), lastExprCode.value());
+      } else {
+        return new ExprCode(code);
```

Comment:
This may cause `Reference` expr value lost.

how aboutting add a if branch at NPE location instead :

```java
 if (last == null) {
        return new ExprCode(null, null, null);
      }
```